apis/duck/v1alpha1/channelable_types.go

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/knative/pkg/apis/duck"
+)
+
+// Channelable is the schema for the channelable portion of the spec
+// section of the resource.
+type Channelable struct {
+	// TODO: What is actually required here for Channel spec.
+	// This is the list of subscriptions for this channel.
+	Subscribers []ChannelSubsciberSpec `json:"subscribers,omitempty"`
+}
+
+// ChannelSubscriberSpec defines a single subscriber to a Channel.
+// TODO: I think the subscriber contract should be Sinkable. You either
+// take it on as your pboblem to deal with, or you reject it. I think
+// a subscription should have no knowledge of what happens down the line
+// and if there are calls, or results follwoing on. You subsribe to me
+// and I give you events and you deal with them??
+// of Call or Result must be present.
+type ChannelSubscriberSpec struct {
```

Comment:
You do this for `Subscribable -> Channelable` below, so `Channelable -> Sinkable` should probably follow the same pattern?

Reply:
No, because I do not want Channels to have to worry about resolving. My thinking was that Channel just gets an endpoint and it does not do any resolving. So this would be domainInternal (or something like that) which gets put in here. Or maybe I misunderstand what you mean here.